build/launch_training.py

```diff
@@ -53,78 +61,124 @@ def main():
-            full_checkpoint_dir = os.path.join(
-                training_args.output_dir, lora_checkpoint_dir
+            sft_trainer.train(
+                model_args=model_args,
+                data_args=data_args,
+                train_args=training_args,
+                peft_config=tune_config,
+                tracker_configs=tracker_config_args,
             )
-
-            logging.info(
-                "Merging lora tuned checkpoint %s with base model into output path: %s",
-                lora_checkpoint_dir,
-                export_path,
+        except MemoryError:
+            logging.error(traceback.format_exc())
+            write_termination_log("OOM error during training")
+            sys.exit(INTERNAL_ERROR_EXIT_CODE)
+        except FileNotFoundError as e:
+            logging.error(traceback.format_exc())
+            write_termination_log("Unable to load file: {}".format(e))
+            sys.exit(USER_ERROR_EXIT_CODE)
+        except (TypeError, ValueError, EnvironmentError) as e:
+            logging.error(traceback.format_exc())
+            write_termination_log(
+                f"Exception raised during training. This may be a problem with your input: {e}"
             )
+            sys.exit(USER_ERROR_EXIT_CODE)
+        except Exception as e:  # pylint: disable=broad-except
+            logging.error(traceback.format_exc())
+            write_termination_log("Unhandled exception during training")
+            sys.exit(INTERNAL_ERROR_EXIT_CODE)
 
-            create_merged_model(
-                checkpoint_models=full_checkpoint_dir,
-                export_path=export_path,
-                base_model=model_args.model_name_or_path,
-                save_tokenizer=True,
-            )
+        if merge_model:
+            try:
+                export_path = os.getenv(
+                    "LORA_MERGE_MODELS_EXPORT_PATH", original_output_dir
+                )
+
+                # get the highest checkpoint dir (last checkpoint)
+                lora_checkpoint_dir = get_highest_checkpoint(training_args.output_dir)
+                full_checkpoint_dir = os.path.join(
+                    training_args.output_dir, lora_checkpoint_dir
+                )
+
+                logging.info(
+                    "Merging lora tuned checkpoint %s with base model into output path: %s",
+                    lora_checkpoint_dir,
+                    export_path,
+                )
+
+                create_merged_model(
+                    checkpoint_models=full_checkpoint_dir,
+                    export_path=export_path,
+                    base_model=model_args.model_name_or_path,
+                    save_tokenizer=True,
+                )
+            except Exception as e:  # pylint: disable=broad-except
+                logging.error(traceback.format_exc())
+                write_termination_log("Exception encountered merging model checkpoints")
```

Comment:
nit: slightly more accurate message
```suggestion
                write_termination_log("Exception encountered merging base model with checkpoint")
```